### ISSUE TITLE
PGO-style mode automatic dynamic shapes

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#120080]
### 🐛 Describe the bug

It's annoying that if you have a model with some dynamism and you're using automatic dynamic shapes, you have to useless compile a static config first, before you discover the real dynamic.

You can workaround this by manually annotating inputs as dynamic or not.

However, if you have some sort of PGO style setup, where you have profiles from previous runs, you can bypass the recompile and the manual annotation, and just use the profile from the previous run to immediately go straight to being dynamic.

We need some sort of PGO setup as a prerequisite. It is probably not worth setting it up for only this. But if we have it, this is a very easy thing to use it for.

### Versions

main

**Comment by masnesral** [comment]
> It is probably not worth setting it up for only this.

Taking the cue that this is not hi-pri, so marking as triaged to remove from the oncall queue.

**Comment by ezyang** [comment]
Yes. Though it does overlap with https://github.com/pytorch/pytorch/pull/130935 this is still a thing we could do, if we wanted to
